### PR TITLE
add optimize for method on MXNet for future optimization

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

This change is intended to bring the accelerator to the DJL MXNet project. User can add addtional options to set the accelerator.

### CODE REVIEW

File: mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @BeforeTest
+    public void downloadSymbol() throws IOException {
+        String url =
+                "https://djl-ai.s3.amazonaws.com/mlrepo/model/cv/image_classification/ai/djl/mxnet/resnet/0.0.1/resnet18_v1-symbol.json";
+        DownloadUtils.download(url, "build/symbol/resnet18_v1-symbol.json");
+    }
+
+    @Test
+    public void testOptimizedFor() {
```
Comment (frankfliu) [inline mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java:33]: String path = "../mxnet/mxnet-model-zoo/src/test/resources/mlrepo/model/cv/image_classification/ai/djl/mxnet/resnet/0.0.1/resnet50_v1-symbol.json";

File: mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import ai.djl.ndarray.NDManager;
+import ai.djl.training.util.DownloadUtils;
+import java.io.IOException;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+public class MxBackendOptimizationTest {
+
+    @BeforeTest
+    public void downloadSymbol() throws IOException {
```
Comment (frankfliu) [inline mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java:26]: We have this file in the repo, we don't need download it

File: mxnet/mxnet-engine/src/main/java/ai/djl/mxnet/engine/MxModel.java [hunk 3]
```diff
@@ -112,6 +113,21 @@ public void load(Path modelPath, String prefix, Map<String, ?> options)
         }
         loadParameters(paramFile, options);
         // TODO: Check if Symbol has all names that params file have
+        // load extra MXNet library
+        if (options != null && options.containsKey("MxloadExtraLib")) {
+            String[] files = ((String) options.get("MxloadExtraLib")).split(",");
```
[the remark this answers is not in the document]
Comment (lanking520) [reply mxnet/mxnet-engine/src/main/java/ai/djl/mxnet/engine/MxModel.java:118]: I don't think so, this is the only way user pass in an option and load the so file

**Review by stu1130 — APPROVED** [review]

**Review by stu1130 — DISMISSED** [review]
The rest LGTM. just change MxLoadExtraLib to environment variable or VM option

File: mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import ai.djl.mxnet.engine.MxNDManager;
+import ai.djl.mxnet.engine.Symbol;
+import ai.djl.ndarray.NDManager;
+import org.testng.annotations.Test;
+
+public class MxBackendOptimizationTest {
+
+    @Test
+    public void testOptimizedFor() {
```
Comment (stu1130) [inline mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java:23]: set environment variable here?

Comment (lanking520) [reply mxnet/mxnet-engine/src/test/java/ai/djl/mxnet/integration/MxBackendOptimizationTest.java:23]: We don't have a lib to load in our test cases.

**Review by stu1130 — APPROVED** [review]

**Review by frankfliu — APPROVED** [review]

**Pull request merged**